extensions/sql-database-projects/src/dialogs/deployDatabaseQuickpick.ts

```diff
@@ -0,0 +1,142 @@
+
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { AppSettingType, IDeployProfile, ILocalDbSetting } from '../models/deploy/deployProfile';
+import { Project } from '../models/project';
+import * as generator from 'generate-password';
+import { getPublishDatabaseSettings } from './publishDatabaseQuickpick';
+let path = require('path');
+let fse = require('fs-extra');
+
+/**
+ * Create flow for Deploying a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchDeployDatabaseQuickpick(project: Project): Promise<IDeployProfile | undefined> {
+
+	// Show options to user for deploy to existing server or docker
+
+	const deployOption = await vscode.window.showQuickPick(
+		[constants.deployToExistingServer, constants.deployToDockerContainer],
+		{ title: constants.selectDeployOption, ignoreFocusOut: true });
+
+	// Return when user hits escape
+	if (!deployOption) {
+		return undefined;
+	}
+
+	let localDbSetting: ILocalDbSetting | undefined;
+	// Deploy to docker selected
+	if (deployOption === constants.deployToDockerContainer) {
+		let portNumber = await vscode.window.showInputBox({
+			title: constants.enterPortNumber,
+			ignoreFocusOut: true,
+			value: constants.defaultPortNumber,
+			validateInput: input => isNaN(+input) ? constants.portMustNotBeNumber : undefined
+		}
+		);
+
+		// Return when user hits escape
+		if (!portNumber) {
+			return undefined;
+		}
+
+		let password: string | undefined = generator.generate({
+			length: 10,
+			numbers: true,
+			symbols: true,
+			lowercase: true,
+			uppercase: true,
+			exclude: '`"\'' // Exclude the chars that cannot be included in the password. Some chars can make the command fail in the terminal
+		});
+		password = await vscode.window.showInputBox({
+			title: constants.enterPassword,
+			ignoreFocusOut: true,
+			value: password,
+			password: true
+		}
+		);
+
+		// Return when user hits escape
+		if (!password) {
+			return undefined;
+		}
+
+		localDbSetting = {
+			serverName: 'localhost',
+			userName: 'sa',
+			dbName: project.projectFileName,
+			password: password,
+			port: +portNumber,
+		};
+	}
+	let deploySettings = await getPublishDatabaseSettings(project, deployOption !== constants.deployToDockerContainer);
+
+	// Return when user hits escape
+	if (!deploySettings) {
+		return undefined;
+	}
+
+	// TODO: Ask for SQL CMD Variables or profile
+
+	let envVarName: string | undefined = '';
+	const integrateWithAzureFunctions: boolean = true; //TODO: get value from settings or quickpick
+
+	//TODO: find a better way to find if AF or local settings is in the project
+	//
+	const localSettings = path.join(project.projectFolderPath, constants.azureFunctionLocalSettingsFileName);
+	const settingExist: boolean = fse.existsSync(localSettings);
+	if (integrateWithAzureFunctions && settingExist) {
+
+		// Ask user to update app settings or not
+		//
+		let choices: { [id: string]: boolean } = {};
```

Comment:
FYI - it's a lot easier to just create objects that extend QuickPickItem with any additional properties you want so that the quickpick selection returns that object. See https://github.com/microsoft/azuredatastudio/blob/main/extensions/sql-database-projects/src/dialogs/addDatabaseReferenceQuickpick.ts#L60

Reply:
will improve in new PR